pkg/bgpv1/gobgp/diffstore_mock.go

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gobgp
+
+import (
+	"golang.org/x/exp/maps"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+var _ DiffStore[*runtime.Unknown] = (*mockDiffStore[*runtime.Unknown])(nil)
+
+type mockDiffStore[T runtime.Object] struct {
```

Comment:
nit: this is a fake not a mock. https://testing.googleblog.com/2013/07/testing-on-toilet-know-your-test-doubles.html.

Reply:
Today I learned